libbeat/common/transport/tls.go

```diff
@@ -154,7 +154,7 @@ func tlsDialWith(
 		}
 	}
 
-	if tlsConfig.InsecureSkipVerify {
+	if config.Verification == tlscommon.VerifyNone {
```

Comment:
A few lines below, in the same function you quoted we set the `tls.Config.VerifyConnection`: https://github.com/elastic/beats/blob/3d3452f1cb00944cd965b9e7fc91dda4ecc61913/libbeat/common/transport/tlscommon/tls_config.go#L120

From the Go docs about this function:
> 	// If normal verification fails then the handshake will abort before
	// considering this callback. This callback will run for all connections
	// regardless of InsecureSkipVerify or ClientAuth settings.
	VerifyConnection func(ConnectionState) error


This is the function that does all the TLS validation to `VerifyFull` and `VerifyCertificate`, for `VerifyStrict` this function only validates the pin as the rest is taken care by the standard library.

We do it like that because the normal TLS validation from Go's standard library (`InsecureSkipVerify = false`) does not allow for certificates with empty SNA, while `VerifyFull` and `VerifyCertificate` do.

The "logging bug" happens because [`tlsDialWith`](https://github.com/elastic/beats/blob/0250eef196a054c3c5efd36a946ae5017d0c271d/libbeat/common/transport/tls.go#L133) does not know about the interaction between  `InsecureSkipVerify` and our `VerifyConnection` function.

Reply:
Ah I see now, thanks. We are manually doing verification for the modes between None and Strict.

Add a comment in the code about this and I'm fine with this change.